AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginAuthIAMTests/AWSDataStoreAuthBaseTest.swift

```diff
@@ -203,57 +197,67 @@ extension AWSDataStoreAuthBaseTest {
     /// - Parameter user
     func signIn(user: TestUser?,
                 file: StaticString = #file,
-                line: UInt = #line) {
+                line: UInt = #line) async {
         guard let user = user else {
             XCTFail("Invalid user", file: file, line: line)
             return
         }
-        let signInInvoked = expectation(description: "sign in completed")
-        _ = Amplify.Auth.signIn(username: user.username,
-                                password: user.password, options: nil) { result in
-            switch result {
-            case .failure(let error):
-                XCTFail("Signin failure \(error)", file: file, line: line)
-                signInInvoked.fulfill() // won't count as pass
-            case .success:
-                signInInvoked.fulfill()
+        let signInInvoked = AsyncExpectation(description: "sign in completed")
+        do {
+            _ = try await Amplify.Auth.signIn(username: user.username,
```

Comment:
From what I understood, you're supposed to wrap these `await` calls inside a `Task`:
```
let anExpectation = asyncExpectation("...")
Task { 
    do {
        // Call some Amplify async API
        await anExpectation.fulfill()
        // All good 👍
    } catch {
        XCTFail("Oh no 🫢! Anyway...")
        await anExpectation.fulfill()
    }
}
await waitForExpectations([anExpectation], timeout: aTimeout)
```

Otherwise if that invocation hangs, then the next lines of codes will never be executed, right?

Reply:
We can certainly add it for calls that we expect to hang.
Do we need it for all the async calls? @brennanMKE

Reply:
I will update this in my next PR.